From de8f1b1fa8c8e592626071916565e3f03da5ec4a Mon Sep 17 00:00:00 2001
From: Jan Chaloupka <jchaloup@redhat.com>
Date: Mon, 20 Jul 2015 13:32:19 +0200
Subject: [PATCH] Update github.com/elazarl/go-bindata-assetfs to at least
 c57a80f1ab2ad67bafa83f5fd0b4c2ecbd253dd5. Otherwise kubernetes does not
 compile with golang 1.5 due to
 https://github.com/elazarl/go-bindata-assetfs/pull/13.

---
 Godeps/Godeps.json                                 |  2 +-
 .../elazarl/go-bindata-assetfs/README.md           | 38 +++++++++++--
 .../elazarl/go-bindata-assetfs/assetfs.go          | 12 +++--
 .../go-bindata-assetfs/go-bindata-assetfs/main.go  | 62 ++++++++++++++++++++++
 4 files changed, 105 insertions(+), 9 deletions(-)
 create mode 100644 Godeps/_workspace/src/github.com/elazarl/go-bindata-assetfs/go-bindata-assetfs/main.go

diff --git a/Godeps/_workspace/src/github.com/elazarl/go-bindata-assetfs/assetfs.go b/Godeps/_workspace/src/github.com/elazarl/go-bindata-assetfs/assetfs.go
index 83cd94d..69d58f8 100644
--- a/Godeps/_workspace/src/github.com/elazarl/go-bindata-assetfs/assetfs.go
+++ b/Godeps/_workspace/src/github.com/elazarl/go-bindata-assetfs/assetfs.go
@@ -3,7 +3,6 @@ package assetfs
 import (
 	"bytes"
 	"errors"
-	"fmt"
 	"io"
 	"io/ioutil"
 	"net/http"
@@ -13,6 +12,10 @@ import (
 	"time"
 )
 
+var (
+	fileTimestamp = time.Now()
+)
+
 // FakeFile implements os.FileInfo interface for a given path and size
 type FakeFile struct {
 	// Path is the path of this file
@@ -37,7 +40,7 @@ func (f *FakeFile) Mode() os.FileMode {
 }
 
 func (f *FakeFile) ModTime() time.Time {
-	return time.Unix(0, 0)
+	return fileTimestamp
 }
 
 func (f *FakeFile) Size() int64 {
@@ -70,6 +73,10 @@ func (f *AssetFile) Readdir(count int) ([]os.FileInfo, error) {
 	return nil, errors.New("not a directory")
 }
 
+func (f *AssetFile) Size() int64 {
+	return f.FakeFile.Size()
+}
+
 func (f *AssetFile) Stat() (os.FileInfo, error) {
 	return f, nil
 }
@@ -98,7 +105,6 @@ func NewAssetDirectory(name string, children []string, fs *AssetFS) *AssetDirect
 }
 
 func (f *AssetDirectory) Readdir(count int) ([]os.FileInfo, error) {
-	fmt.Println(f, count)
 	if count <= 0 {
 		return f.Children, nil
 	}
-- 
1.9.3

